lib/codecs/Base58DidUrlDecoder.js

```diff
@@ -28,13 +27,14 @@ export class Base58DidUrlDecoder extends CborldDecoder {
     return url;
   }
 
-  static createDecoder({value} = {}) {
+  static createDecoder({value, transformer} = {}) {
+    const decompressionMap = transformer.reverseUrlSchemeTable;
     if(!(Array.isArray(value) && value.length > 1 && value.length <= 3)) {
       return false;
     }
-    if(!ID_TO_SCHEME.has(value[0])) {
+    if(!decompressionMap.has(value[0])) {
       return false;
     }
-    return new Base58DidUrlDecoder();
+    return new Base58DidUrlDecoder({decompressionMap});
```

Comment:
Since it won't let me suggest here, here's what I'm thinking:

```js
  const prefix = decompressionMap.get(value[0]);
  if(prefix === undefined) {
    return false;
  }
  return new Base58DidUrlDecoder({prefix});
```

Reply:
I'd think this same sort of pattern could be used in all the places to avoid the double map hits.